client/src/components/UserPrompt.js

```diff
@@ -49,10 +57,28 @@ const UserPrompt = () => {
   return (
     <div>
       <Snackbar
-        open={snackbarOpen}
+        open={
+          ((matchState.turn === "blue" &&
+            matchState.turnPhase === "guess" &&
+            matchState.blueGuessers.findIndex(
+              (player) => player.email === email
+            ) !== -1) ||
+            (matchState.turn === "red" &&
+              matchState.turnPhase === "guess" &&
+              matchState.redGuessers.findIndex(
+                (player) => player.email === email
+              ) !== -1) ||
+            (matchState.turn === "blue" &&
+              matchState.turnPhase === "clue" &&
+              matchState.blueSpymaster.email === email) ||
+            (matchState.turn === "red" &&
+              matchState.turnPhase === "clue" &&
+              matchState.redSpymaster.email === email)) && snackbarOpen
```

Comment:
seems redundant with what is happening in useEffect, since these checks are already happening there.

Reply:
yes you're right I'm checking for the user role again but I can just use the state, I changed it now